File: solutions/0020-valid-parentheses/solution.cpp
class Solution {
public:
    bool isValid(string s) {
        while (s.size() > 0) {
            std::cout << s << std::endl;
            if (s.find("()") != -1) {
                s.erase(s.begin() + s.find("()"), s.begin() + s.find("()") + 2);
            } else if (s.find("{}") != -1) {
                s.erase(s.begin() + s.find("{}"), s.begin() + s.find("{}") + 2);
            } else if (s.find("[]") != -1) {
                s.erase(s.begin() + s.find("[]"), s.begin() + s.find("[]") + 2);
            } else {
                return false;
            }
        }
        return true;
    }
};


// (([])){}

File: solutions/0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c == ')' or c == ']' or c == '}':
                if len(stack) != 0:
                    if stack[-1] == '(' and c == ')':
                        stack.pop()
                    elif stack[-1] == '[' and c == ']':
                        stack.pop()
                    elif stack[-1] == '{' and c == '}':
                        stack.pop()
                    else:
                        return False
                else:
                    return False
            else:
                stack.append(c)
        if len(stack) == 0:
            return True
        return False
            
